### ISSUE TITLE
Build container images in parallel?

### ISSUE THREAD

**Issue by psiroky** [CentOS-PaaS-SIG/contra-env-setup#46]
Hello,
IIUC correctly all the container images are currently build serially. Would it make sense to queue all the builds in parallel and then just wait for all of them to finish? This should speed up the setup quite a bit. Also, in lot of cases the build are IO bound, not CPU bound, so this should not even require more CPU power or so.

**Comment by arilivigni** [comment]
@psiroky I think it is a nice to have since you don't usually build everything all over again.  You do life cycle management of the images through OpenShift.  That said if you wanted to implement that strategy I am not opposed to it.  The other thing besides building is waiting for Jenkins to be up and running which we don't do today.

**Comment by psiroky** [comment]
I agree it's definitely nice to have. However, if we want to do automated testing for contra-env-setup, then we will be building stuff from scratch for each test build (at least that was my idea for e.g. CVP as we want to test things with clean environment). Let's see if this becomes issue for us or not. If yes I could take a look at implementing this.

**Comment by arilivigni** [comment]
@psiroky great point and we do want that so parallel building would be ideal.  Is this something you can work on possibly?

**Comment by psiroky** [comment]
@arilivigni yeah, looks like a good opportunity to dive a bit more the into Ansible world. I'll try to find some time in coming weeks to tackle this.

**Comment by arilivigni** [comment]
https://github.com/CentOS-PaaS-SIG/contra-env-setup/pull/59
